# Final Project - Disney Box Office Analysis

# Introduction

## Questions of Interest
In this analysis, I will be proposing a series of questions regarding Disney's Box Office and answering them to the best of my abilities. The questions are: 
1. Which year had the highest total Box Office (measured with both Total Gross and Inflation-Adjusted Gross)? 
2. Which genre had the highest total Box Office (measured with both Total Gross and Inflation-Adjusted Gross)? 
3. Which director had the highest total Box Office (measured with both Total Gross and Inflation-Adjusted Gross)? 
4. What is the most successful Disney movie ever made Box-Office-wise adjusted for inflation? 

As a filmmaker and movie-lover myself, it would be interesting to find out how this artistic medium I adore so much has evolved, and which movie/director/genre still reined supreme, through through the lens of Disney. I suppose that the **Action** genre would take the cake as the highest Box Office earner, but I would not be surprised if the **Musical** genre achieves a surprise upset. Let's see if my bold prediction is accurate or not! 

# Dataset Description

The dataset I will be using, Disney-related and located in the 'data' folder within the Jupyter notebook server, has been provided to me by the fantastic people of running the program; from it, I will use two tables, both **csv files**, and they are listed below with a breif description: 

* **disney_movies_total_gross.csv** 
    * This file contains the movie title, its release date, genre, MPAA rating, total gross, and inflation adjusted gross. 
    
* **disney-director.csv**
    * This file contains the movie title, and the name of its director. 

# Method and Results

Let's import the necessary libraries and tables first before moving on to sorting/cleaning up the data! 

In [1]:
# importing the libraries needed
import pandas as pd
import altair as alt

# importing the files we need
# parsing the date for the 'movie' file since we'll need it later
movie = pd.read_csv('data/disney_movies_total_gross.csv', parse_dates = ['release_date'])
director = pd.read_csv('data/disney-director.csv')

Let's see what both tables look like.

In [2]:
movie.head()

movie_title release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                        Pinocchio   1940-02-09  Adventure           G   
2                         Fantasia   1940-11-13    Musical           G   
3                Song of the South   1946-11-12  Adventure           G   
4                       Cinderella   1950-02-15      Drama           G   

    total_gross inflation_adjusted_gross  
0  $184,925,485           $5,228,953,251  
1   $84,300,000           $2,188,229,052  
2   $83,320,000           $2,187,090,808  
3   $65,000,000           $1,078,510,579  
4   $85,000,000             $920,608,730

In [3]:
director.head()

name        director
0  Snow White and the Seven Dwarfs      David Hand
1                        Pinocchio  Ben Sharpsteen
2                         Fantasia    full credits
3                            Dumbo  Ben Sharpsteen
4                            Bambi      David Hand

Let's get some other information of both tables by using '.info()'!

In [4]:
movie.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 579 entries, 0 to 578
Data columns (total 6 columns):
 #   Column                    Non-Null Count  Dtype         
---  ------                    --------------  -----         
 0   movie_title               579 non-null    object        
 1   release_date              579 non-null    datetime64[ns]
 2   genre                     562 non-null    object        
 3   MPAA_rating               523 non-null    object        
 4   total_gross               579 non-null    object        
 5   inflation_adjusted_gross  579 non-null    object        
dtypes: datetime64[ns](1), object(5)
memory usage: 27.3+ KB


The 'movie' table includes 579 rows and 6 columns in total. Every **movie_title** has its **release date**, **genre**, **MPAA_rating**, **total_gross**, and **inflation_adjusted_gross** attached. 

Let's see what the 'director' file includes!

In [5]:
director.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 56 entries, 0 to 55
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   name      56 non-null     object
 1   director  56 non-null     object
dtypes: object(2)
memory usage: 1.0+ KB


The 'director' table contains much less information by contrast, with only movie **name** and **director** included. The file contains 56 entries and 2 columns.

Before we can get to answering the proposed questions, let's first do some filtering and prep! 

First, let's add another column to the 'movie' dataset showcasing the 'year' of each movie's release, titled 'release_year'.

In [6]:
movie = movie.assign(release_year = movie['release_date'].dt.year)
movie.head()

movie_title release_date      genre MPAA_rating  \
0  Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                        Pinocchio   1940-02-09  Adventure           G   
2                         Fantasia   1940-11-13    Musical           G   
3                Song of the South   1946-11-12  Adventure           G   
4                       Cinderella   1950-02-15      Drama           G   

    total_gross inflation_adjusted_gross  release_year  
0  $184,925,485           $5,228,953,251          1937  
1   $84,300,000           $2,188,229,052          1940  
2   $83,320,000           $2,187,090,808          1940  
3   $65,000,000           $1,078,510,579          1946  
4   $85,000,000             $920,608,730          1950

In the **'movie.info'** return, we can see that both 'total_gross' and 'inflation_adjusted_gross' are of the **'object'** column type, which is not what we want at all. Instead, we need to convert them both to **'integer'** type. 

To achieve this, we must first remove the **'$'** and **','** symbols from the tables!

In [7]:
# delete the '$' and ',' symbols from total_gross and inflation_adjusted_gross 
movie['total_gross'] = movie['total_gross'].str.replace('$','')
movie['total_gross'] = movie['total_gross'].str.replace(',','')
movie['inflation_adjusted_gross'] = movie['inflation_adjusted_gross'].str.replace('$','')
movie['inflation_adjusted_gross'] = movie['inflation_adjusted_gross'].str.replace(',','')

movie

<ipython-input-7-fc6ee8f08b85>:2: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will*not* be treated as literal strings when regex=True.
  movie['total_gross'] = movie['total_gross'].str.replace('$','')
<ipython-input-7-fc6ee8f08b85>:4: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will*not* be treated as literal strings when regex=True.
  movie['inflation_adjusted_gross'] = movie['inflation_adjusted_gross'].str.replace('$','')


movie_title release_date      genre MPAA_rating  \
0    Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                          Pinocchio   1940-02-09  Adventure           G   
2                           Fantasia   1940-11-13    Musical           G   
3                  Song of the South   1946-11-12  Adventure           G   
4                         Cinderella   1950-02-15      Drama           G   
..                               ...          ...        ...         ...   
574         The Light Between Oceans   2016-09-02      Drama       PG-13   
575                   Queen of Katwe   2016-09-23      Drama          PG   
576                   Doctor Strange   2016-11-04  Adventure       PG-13   
577                            Moana   2016-11-23  Adventure          PG   
578     Rogue One: A Star Wars Story   2016-12-16  Adventure       PG-13   

    total_gross inflation_adjusted_gross  release_year  
0     184925485               5228953251          1937  
1      84300000               2188229052          1940  
2      83320000               2187090808          1940  
3      65000000               1078510579          1946  
4      85000000                920608730          1950  
..          ...                      ...           ...  
574    12545979                 12545979          2016  
575     8874389                  8874389          2016  
576   232532923                232532923          2016  
577   246082029                246082029          2016  
578   529483936                529483936          2016  

[579 rows x 7 columns]

With the two symbols out of the way, we can now safely transfrom the two columns from **object** type to **integer** without encountering any errors!

In [8]:
# changing column type for 'total_gross'
movie = movie.assign(total_gross = movie['total_gross'].astype(int))

# changing column type by first transforming it into 'str', then to 'int'
movie['inflation_adjusted_gross'] = movie['inflation_adjusted_gross'].astype(str).astype(int)
movie

movie_title release_date      genre MPAA_rating  \
0    Snow White and the Seven Dwarfs   1937-12-21    Musical           G   
1                          Pinocchio   1940-02-09  Adventure           G   
2                           Fantasia   1940-11-13    Musical           G   
3                  Song of the South   1946-11-12  Adventure           G   
4                         Cinderella   1950-02-15      Drama           G   
..                               ...          ...        ...         ...   
574         The Light Between Oceans   2016-09-02      Drama       PG-13   
575                   Queen of Katwe   2016-09-23      Drama          PG   
576                   Doctor Strange   2016-11-04  Adventure       PG-13   
577                            Moana   2016-11-23  Adventure          PG   
578     Rogue One: A Star Wars Story   2016-12-16  Adventure       PG-13   

     total_gross  inflation_adjusted_gross  release_year  
0      184925485                5228953251          1937  
1       84300000                2188229052          1940  
2       83320000                2187090808          1940  
3       65000000                1078510579          1946  
4       85000000                 920608730          1950  
..           ...                       ...           ...  
574     12545979                  12545979          2016  
575      8874389                   8874389          2016  
576    232532923                 232532923          2016  
577    246082029                 246082029          2016  
578    529483936                 529483936          2016  

[579 rows x 7 columns]

Next, let's drop the **MPAA_rating** column as it is irrelevant to our analysis. 

In [9]:
movie = movie.drop('MPAA_rating', axis = 1)
movie

movie_title release_date      genre  total_gross  \
0    Snow White and the Seven Dwarfs   1937-12-21    Musical    184925485   
1                          Pinocchio   1940-02-09  Adventure     84300000   
2                           Fantasia   1940-11-13    Musical     83320000   
3                  Song of the South   1946-11-12  Adventure     65000000   
4                         Cinderella   1950-02-15      Drama     85000000   
..                               ...          ...        ...          ...   
574         The Light Between Oceans   2016-09-02      Drama     12545979   
575                   Queen of Katwe   2016-09-23      Drama      8874389   
576                   Doctor Strange   2016-11-04  Adventure    232532923   
577                            Moana   2016-11-23  Adventure    246082029   
578     Rogue One: A Star Wars Story   2016-12-16  Adventure    529483936   

     inflation_adjusted_gross  release_year  
0                  5228953251          1937  
1                  2188229052          1940  
2                  2187090808          1940  
3                  1078510579          1946  
4                   920608730          1950  
..                        ...           ...  
574                  12545979          2016  
575                   8874389          2016  
576                 232532923          2016  
577                 246082029          2016  
578                 529483936          2016  

[579 rows x 6 columns]

As indicated in the 'info' result, the 'movie' table does include some null values. Unfortunately, there is no method to ascertain the appropriate genre for each of those entries with the data at hand, therefore, I intend to replace those null values with **N/A** and lump them into the same category.

In [10]:
movie['genre'] = movie['genre'].fillna('N/A')

# the code below searches for and displays the entries with the 'N/A' value
movie[movie.eq('N/A').any(1)]

movie_title release_date genre  total_gross  \
20   The Many Adventures of Winnie the Pooh   1977-03-11   N/A            0   
22               Herbie Goes to Monte Carlo   1977-06-24   N/A     28000000   
23                           The Black Hole   1979-12-21   N/A     35841901   
24                         Midnight Madness   1980-02-08   N/A      2900000   
25            The Last Flight of Noah’s Ark   1980-06-25   N/A     11000000   
26                 The Devil and Max Devlin   1981-01-01   N/A     16000000   
121                                 Newsies   1992-04-08   N/A      2706352   
122                             Passed Away   1992-04-24   N/A      4030793   
128            A Gun in Betty Lou's Handbag   1992-08-21   N/A      3591460   
146                          Bound by Honor   1993-04-16   N/A      4496583   
155                     My Boyfriend's Back   1993-08-06   N/A      3218882   
156                             Father Hood   1993-08-27   N/A      3268203   
168                           Red Rock West   1994-01-28   N/A      2502551   
251                         The War at Home   1996-11-20   N/A        34368   
304                               Endurance   1999-05-14   N/A       229128   
350                High Heels and Low Lifes   2001-10-26   N/A       226792   
355                     Frank McKlusky C.I.   2002-01-01   N/A            0   

     inflation_adjusted_gross  release_year  
20                          0          1977  
22                  105847527          1977  
23                  120377374          1979  
24                    9088096          1980  
25                   34472116          1980  
26                   48517980          1981  
121                   5497481          1992  
122                   8187848          1992  
128                   7295423          1992  
146                   9156084          1993  
155                   6554384          1993  
156                   6654819          1993  
168                   5170709          1994  
251                     65543          1996  
304                    380218          1999  
350                    337782          2001  
355                         0          2002

With both datasets now prepared, we can now move onto creating the two functions necessary to run a thorough analysis with. The first function, called **biggest_box_office**, is designed to output a singular row with the best gross, either total or inflation-adjusted, based on the data input. 

In [11]:
from functions_box_office import biggest_box_office

The second function, titled **box_office_barChart**, is used to produce a bar chart based on the various axis and label input of the same chosen data. 

In [12]:
from functions_box_office import box_office_barChart

With all the ground work done, the first question can now be answered. 

Which year had the highest total Box Office (measured with both Total Gross and Inflation-Adjusted Gross)? 

In [13]:
year_box_office = movie.groupby(by = 'release_year').sum().reset_index()

best_year_total_gross = biggest_box_office(year_box_office, 'total_gross')
best_year_total_gross

release_year  total_gross  inflation_adjusted_gross
53          2016   2873393105                2873393105

As the result has shown, **2016** had the best box office gross, of **$2,873,393,105** without adjusted for inflation, but, you may ask, how does that year compare to other year? Let's put it through the bar chart and find out!

In [14]:
year_total_barChart = box_office_barChart(year_box_office, 'release_year', 'total_gross', 
                                          'Release Year', 'Total Gross', 'Total Gross based on Release Year')
year_total_barChart

alt.Chart(...)

The result is staggeringly one-sided, which is perhaps not that surprising given how much more peaceful the 80s and onwards has been compared to prior. Let's see if the result is still the same after adjusting for inflation!

In [15]:
best_year_inflation_gross = biggest_box_office(year_box_office, 'inflation_adjusted_gross')
best_year_inflation_gross

release_year  total_gross  inflation_adjusted_gross
0          1937    184925485                5228953251

After adjusting for inflation, we have now went back in time to **1937** with an astonishing amount of **$5,228,953,251** box office gross in total that year! Inflation certainly plays a big part in changing our perception of value. Let's see how it looks in bar chart!

In [16]:
year_inflation_barChart = box_office_barChart(year_box_office, 'release_year', 'inflation_adjusted_gross', 
                                              'Release Year', 'Inflation-adjusted Gross', 
                                              'Inflation-adjusted Gross based on Release Year')

year_inflation_barChart

alt.Chart(...)

It seems like 1937 and 1940 still are still way ahead conpared to other years. The 2000s and onwards look to have sustained a considerably growth, but none could still come close to those two extremely profitable years!

Moving onto Qesution 2, which asks, which genre has had the biggest box office success compared to others? 

First, let's find the winner based on total gross!

In [17]:
genre_box_office = movie.groupby(by = 'genre').sum().drop('release_year', axis=1).reset_index()

best_genre_total_gross = biggest_box_office(genre_box_office, 'total_gross')

best_genre_total_gross

genre  total_gross  inflation_adjusted_gross
1  Adventure  16389069453               24561266158

It seems that my initial prediction was quite off the mark with **Adventure** taking the throne instead with a staggering **$16,389,069,453**. Let's plot the bar chart and see how ahead this genre is!

In [18]:
genre_total_barChart = box_office_barChart(genre_box_office, 'genre', 'total_gross', 
                    'Genre', 'Total Gross', 'Total Gross based on Genre')

genre_total_barChart

alt.Chart(...)

**Adventure** takes the lead by a long shot and almost quadruples the **Action** genre. Surprisingly, **Comedy** takes second place overshadows my initial prediction almost by doubling the total gross. 

With **Total Gross** done, now let's look at which genre reigns supreme once adjusted for inflation!

In [19]:
best_genre_inflation_gross = biggest_box_office(genre_box_office, 'inflation_adjusted_gross')

best_genre_inflation_gross

genre  total_gross  inflation_adjusted_gross
1  Adventure  16389069453               24561266158

The result seems to be the same as **Total Gross**, which is unsurprising due to how much the **Adventure** genre dwarfs others. Let's graph it out and see just how dominant our winner is!

In [20]:
genre_inflation_gross = box_office_barChart(genre_box_office, 'genre', 'inflation_adjusted_gross', 
                        'Genre', 'Inflation_adjusted Gross', 'Inflation_adjusted Gross based on Genre')

genre_inflation_gross

alt.Chart(...)

Although the two graphs, **Total Gross** and **Inflation-adjusted Gross**, look quite the same, the latter does include quite a big jump for the **Drama** and **Musical** genres, as those two were once the main productions and cash cows of the golden age of Hollywood!

Similar to both Question 1 and 2, Question 3 asks which director is the most commericially successful based on their movies' box office results? 

Let's first merge the two dataframes **movie** and **director** together before continuing onwards!

In [21]:
movie_and_director = movie.merge(director, left_on = 'movie_title', right_on = 'name', how = 'inner')
movie_and_director.drop(columns = 'name')

movie_title release_date      genre  \
0          Snow White and the Seven Dwarfs   1937-12-21    Musical   
1                                Pinocchio   1940-02-09  Adventure   
2                                 Fantasia   1940-11-13    Musical   
3                               Cinderella   1950-02-15      Drama   
4                               Cinderella   2015-03-13      Drama   
5                       Lady and the Tramp   1955-06-22      Drama   
6                          Sleeping Beauty   1959-01-29      Drama   
7                           101 Dalmatians   1961-01-25     Comedy   
8                           101 Dalmatians   1996-11-27     Comedy   
9                   The Sword in the Stone   1963-12-25  Adventure   
10                         The Jungle Book   1967-10-18    Musical   
11                         The Jungle Book   1994-12-25  Adventure   
12                         The Jungle Book   2016-04-15  Adventure   
13                          The Aristocats   1970-04-24    Musical   
14  The Many Adventures of Winnie the Pooh   1977-03-11        N/A   
15                            The Rescuers   1977-06-22  Adventure   
16                   The Fox and the Hound   1981-07-10     Comedy   
17                      The Black Cauldron   1985-07-24  Adventure   
18               The Great Mouse Detective   1986-07-02  Adventure   
19                        Oliver & Company   1988-11-18  Adventure   
20                      The Little Mermaid   1989-11-15  Adventure   
21                 The Rescuers Down Under   1990-11-16  Adventure   
22                    Beauty and the Beast   1991-11-13    Musical   
23                                 Aladdin   1992-11-11     Comedy   
24                           The Lion King   1994-06-15  Adventure   
25                              Pocahontas   1995-06-10  Adventure   
26             The Hunchback of Notre Dame   1996-06-21  Adventure   
27                                Hercules   1997-06-13  Adventure   
28                                   Mulan   1998-06-19  Adventure   
29                                  Tarzan   1999-06-16  Adventure   
30                                Dinosaur   2000-05-19  Adventure   
31                The Emperor's New Groove   2000-12-15  Adventure   
32               Atlantis: The Lost Empire   2001-06-08  Adventure   
33                           Lilo & Stitch   2002-06-21  Adventure   
34                         Treasure Planet   2002-11-27  Adventure   
35                            Brother Bear   2003-10-24  Adventure   
36                       Home on the Range   2004-04-02     Comedy   
37                          Chicken Little   2005-11-04  Adventure   
38                      Meet the Robinsons   2007-03-30  Adventure   
39                                    Bolt   2008-11-21     Comedy   
40               The Princess and the Frog   2009-11-25  Adventure   
41                     Alice in Wonderland   2010-03-05  Adventure   
42                                 Tangled   2010-11-24  Adventure   
43                         Winnie the Pooh   2011-07-15  Adventure   
44                          Wreck-It Ralph   2012-11-02  Adventure   
45                                  Frozen   2013-11-22  Adventure   
46                              Big Hero 6   2014-11-07  Adventure   
47                                Zootopia   2016-03-04  Adventure   
48                                   Moana   2016-11-23  Adventure   

    total_gross  inflation_adjusted_gross  release_year             director  
0     184925485                5228953251          1937           David Hand  
1      84300000                2188229052          1940       Ben Sharpsteen  
2      83320000                2187090808          1940         full credits  
3      85000000                 920608730          1950      Wilfred Jackson  
4     201151353                 201151353          2015      Wilfred Jackson  
5      93600000                1236035515          1955 

With the now-combined **movie_and_director** table, let's filter and find out who has made the best box office returns based on **Total Gross**?

In [22]:
best_director_total_gross = biggest_box_office(movie_and_director, 'total_gross')
best_director_total_gross

movie_title release_date      genre  total_gross  \
24  The Lion King   1994-06-15  Adventure    422780140   

    inflation_adjusted_gross  release_year           name      director  
24                 761640898          1994  The Lion King  Roger Allers

Given how iconic this movie, perhaps it is only right for **The Lion King** to announce itself as the king of all Disney films (based on **Total Gross**), as the 1994 classics have ingrained itself into many people's childhood! 

Let's look at the graph and see how each director stacks up against each other; some directors have directed multiple Disney projects, as indicated in the **movie_and_director** data set, therefore, the results should be a bit lopsided.

In [23]:
director_box_office = movie_and_director.groupby(by = 'director').sum().drop('release_year', axis=1).reset_index()

director_total_barChart = box_office_barChart(director_box_office, 'director', 'total_gross', 
                                               'director', 'Total Gross', 'Total Gross based on Director')

director_total_barChart

alt.Chart(...)

With **Wolfgang Reitherman** having directed an incredible amount of **9 Disney movies** and accumulating over **9 hundred million dollars**, no other director can be crowned for this achievement! 

But can he still sit comfortably on his throne after being adjusted for inflation? Let's find out!

In [24]:
best_director_inflation_gross = biggest_box_office(movie_and_director, 'inflation_adjusted_gross')
best_director_inflation_gross

movie_title release_date    genre  total_gross  \
0  Snow White and the Seven Dwarfs   1937-12-21  Musical    184925485   

   inflation_adjusted_gross  release_year                             name  \
0                5228953251          1937  Snow White and the Seven Dwarfs   

     director  
0  David Hand

As the one that started it all, **Snow White and the Seven Dwarfs** is still the unquestionable queen of all animations and Disney movies with the inflation-adjusted gross of **5,228,953,251**! 

Let's put together the bar chart with inflation-adjusted gross and see if **Wolfgang Reitherman** is still the rightful king!

In [25]:
director_inflation_gross = box_office_barChart(director_box_office, 'director', 'inflation_adjusted_gross', 
                                               'director', 'Inflation_adjusted Gross', 'Inflation-adjusted Gross based on Director')

director_inflation_gross

alt.Chart(...)

At last, the king, **Wolfgang Reitherman**, has fallen to **David Hand**, as his sole directorial credit out-earns virtually all other directors' total box office returns! An unbelievable amount of over **$5 Billion** is acheived with just one singular movie, **Snow White and the Seven Dwarfs**!

Finally, Question 4 asks which movie holds the biggest box office record for Disney once adjusted for inflation?

In [26]:
movie.sort_values(by = 'inflation_adjusted_gross', ascending = False).head(1)

movie_title release_date    genre  total_gross  \
0  Snow White and the Seven Dwarfs   1937-12-21  Musical    184925485   

   inflation_adjusted_gross  release_year  
0                5228953251          1937

Unsurpisingly, **Snow White and the Seven Dwarfs** takes home the trophy! I personally find this result to be very fitting and satisifying given that this is the first animation Disney has ever made, and the movie that, if failed, would have meant the demise of the company. 

# Discussion

In this report, I used both **disney_movies_total_gross** and **disney_director** datasets and tried to find the movie with the biggest box office results based on various categoris; while answering these questions, I also charted and compared other categorical outputs alongside its winner. Overall, **inflation** has played a big role in altering the results as certain years and directorial works have outshined the rest once their box office returns have been properly adjusted. 

My initial prediction of **Action** taking the throne for "biggest box office based on genre" had been quite off mark, with **Adventure** and **Comedy** beating it as they take both first and second place respectively. However, it is worth noting that a movie, often times, can be categorized into several genres at once, so you may find the outcomes slightly off-mark for you personally. An example of this would be "The Lion King" as it is categorized as "Adventure," even though it could also fit the "Musical" criteria as well. 

One question that could be proposed and looked at would be finding the movie with the biggest box office results based on **MPAA ratings**. It might be worth analyzing to assess which rating may yield the highest return, but given that movies with a higher age restriction usually open to a more limited pool of audience, I had made the decision to axe it since the the results might be too obvious and lack the depth necessary to do a proper analysis on. 

# References

All datasets used are provided to me by the UBC faculty that oversees the Extended Learning Program called "Programming in Python for Data Science."